packages/js/src/website.ts

```diff
@@ -61,45 +65,94 @@ type FormbricksWebsiteMethods = {
-          console.error(
-            "🧱 Formbricks - Global error: You need to call formbricks.init before calling any other method"
-          );
-          return;
-        }
+      const methodQueue = new MethodQueue();
 
-        // still need to check if the apiHost is passed
-        if (!args[0]) {
-          console.error("🧱 Formbricks - Global error: You need to pass the apiHost as the first argument");
+      if (prop === "init") {
+        // if still initializing, return
+        if (isInitializing) {
           return;
         }
 
+        // mark as initializing
+        isInitializing = true;
+
         const { apiHost } = args[0];
         const loadSDKResult = await wrapThrowsAsync(loadFormbricksWebsiteSDK)(apiHost);
 
         if (!loadSDKResult.ok) {
+          // error loading the SDK
+          isInitializing = false;
           console.error(`🧱 Formbricks - Global error: ${loadSDKResult.error.message}`);
           return;
         }
+
+        try {
+          const result = await (
+            (window.formbricks as TFormbricksWebsite)[prop as FormbricksWebsiteMethods] as Function
+          )(...args);
+
+          // mark as initialized
+          isInitializing = false;
+          isInitialized = true;
+
+          // run the method queue
+          methodQueue.run();
+
+          // clear the method queue
+          methodQueue.clear();
+          return result;
+        } catch (error) {
+          isInitializing = false;
+          console.error(`🧱 Formbricks - Global error: ${error}`);
+          throw error;
+        }
       }
 
-      if (window.formbricks && typeof window.formbricks[prop as FormbricksWebsiteMethods] !== "function") {
+      if (!isInitialized) {
+        if (!isInitializing) {
+          console.error(
+            "🧱 Formbricks - Global error: You need to call formbricks.init before calling any other method"
+          );
+          return;
+        }
+
+        return new Promise<void>((resolve) => {
+          const method = async () => {
+            try {
+              const result = await (
+                (window.formbricks as TFormbricksWebsite)[prop as FormbricksWebsiteMethods] as Function
+              )(...args);
+              resolve(result);
+            } catch (error) {
+              console.error(`🧱 Formbricks - Global error: ${error}`);
+            }
+          };
+
+          methodQueue.add(method);
+        });
+      }
+
+      if (
+        window.formbricks &&
+        typeof (window.formbricks as TFormbricksWebsite)[prop as FormbricksWebsiteMethods] !== "function"
+      ) {
         console.error(
           `🧱 Formbricks - Global error: Formbricks Website SDK does not support method ${String(prop)}`
         );
         return;
       }
 
       try {
-        return (window.formbricks[prop as FormbricksWebsiteMethods] as Function)(...args);
+        return ((window.formbricks as TFormbricksWebsite)[prop as FormbricksWebsiteMethods] as Function)(
```

Comment:
shouldn't we also add this to the queue here to avoid that commands are executed in the wrong order?
(e.g. newer request after init before requests that were queued first)